servers/curated-corpus-api/src/admin/resolvers/mutations/SectionItem/index.ts

```diff
@@ -36,11 +39,22 @@ export async function createSectionItem(
     throw new AuthenticationError(ACCESS_DENIED_ERROR);
   }
 
-  const sectionItem = await dbCreateSectionItem(context.db, {
-    approvedItemExternalId: data.approvedItemExternalId,
-    rank: data.rank,
-    sectionId: section.id,
-  });
+  // Determine the source based on the authenticated user
+  // ML Lambda has username === 'ML', all other users are MANUAL
+  const createSource =
+    context.authenticatedUser.username === 'ML'
+      ? ActivitySource.ML
+      : ActivitySource.MANUAL;
```

Comment:
In Slack I proposed adding an argument to report who is trying to create the item, but when diving in, it seemed better to me to rely on the `username`:
1. This makes it easier for ML to make requests; one less parameter to set.
2. This is authenticated, verified information.

Reply:
> would this username ever change?

Good thought about the username being changed! I'll extract `'ML'` into a shared constant to ensure behavior won't break if the username is changed.

> but thinking if it would be better to lookup the sectionItem's section & get the createSource from there? don't have a strong opinion on this

We want to know who’s creating the section item. Right now, only ML can add items to ML sections, but editorial has shown interest in doing so as well. We don’t know yet if that’ll ever happen, but ideally we shouldn’t bake in the assumption that only ML can create items in ML sections.